tasks/notify-slack-on-failure/notify-slack-on-failure.yaml

```
@@ -0,0 +1,74 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: notify-slack-on-failure
+spec:
+  description: >-
+    Sends an error message to Slack using postMessage API if managed pipelines fail
+  params:
+    - name: secretName
+      description: Name of secret which contains authentication token for app
+    - name: secretKeyName
+      description: Name of key within secret which contains webhook URL
+    - name: release
+      description: String representation of Release spec
+  volumes:
+    - name: slack-token
+      secret:
+        secretName: $(params.secretName)
+        optional: true
+  steps:
+    - name: notify-failure
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      volumeMounts:
+        - name: slack-token
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: KEYNAME
+          value: $(params.secretKeyName)
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        if [ -z "${KEYNAME}" ] ; then
+          echo "No secret key name provided via the 'secretKeyName' pipeline parameter \ 
+          provided in the release-plan"
+          echo "No message will be sent to Slack"
+          exit
+        fi
+
+        if [ -f "/etc/secrets/${KEYNAME}" ]; then
+          WEBHOOK_URL=$(cat "/etc/secrets/${KEYNAME}")
+        else
+          echo "Error: Secret not defined properly. The key to use (${KEYNAME}) is defined in the 'secretKeyName' \
+              pipeline parameter provided in the release-plan but the Secret does not contain the key"
```

Similar to another place: This way all the indentation on the second line is included in the echo. So it's better to split it into two strings:
```suggestion
          echo "Error: Secret not defined properly. The key to use (${KEYNAME}) is defined in the 'secretKeyName'" \
              "pipeline parameter provided in the release-plan but the Secret does not contain the key"
```

---

fixed